pkg/providers/instance/instance.go

```diff
@@ -222,7 +221,7 @@ func (p *DefaultProvider) launchInstance(ctx context.Context, nodeClass *v1.EC2N
 		reason, message := awserrors.ToReasonMessage(err)
 		return ec2types.CreateFleetInstance{}, cloudprovider.NewCreateError(fmt.Errorf("getting launch template configs, %w", err), reason, fmt.Sprintf("Error getting launch template configs: %s", message))
 	}
-	if err := p.checkODFallback(nodeClaim, instanceTypes, launchTemplateConfigs); err != nil {
+	if err := p.checkODFallback(ctx, nodeClaim, instanceTypes, launchTemplateConfigs); err != nil {
 		log.FromContext(ctx).Error(err, "failed while checking on-demand fallback")
```

Comment:
Maybe this error message should be updated because it might be confusing -- this shouldn't prevent it from launching OD in this scenario so I don't see a reason for surfacing this as an option if it doesn't tangibly affect the business logic, only the logging